### ISSUE TITLE
FEAT: Allow automatic pass through to children if no parent component is provided

### ISSUE THREAD

**Issue by Soviut** [vuejs/router#1399]
### What problem does this feature solve?
Currently, if you want to use child routes, the parent must have a component. In many cases, I don't need any kind of parent template around the children, so I wind up creating an empty "pass through" component that contains nothing but a `<router-view />`.

Ideally, if a route has children, the parent's `component` property would become optional. Either the parent component simply wouldn't be used, or a pass through component could be automatically injected if keeping the page hierarchy consistent is important.

### What does the proposed API look like?
```typescript
{
  path: '/posts',
  // --- no need for a component here ---
  children: [
    {
      path: '',
      component: PostsIndex,
      name: 'posts'
    }
  ]
}
```

**Comment by posva** [comment]
Duplicate of https://github.com/vuejs/router/issues/1174

**Comment by posva** [comment]
It's already implemented btw, will be out in 4.1
